libtsuba/include/katana/EntityTypeManager.h

```diff
@@ -527,6 +527,30 @@ class KATANA_EXPORT EntityTypeManager {
     return std::nullopt;
   }
 
+  /// \returns the names of the atomic supertypes for the given non-atomic type
+  /// \p entity_type_id is an non-atomic type
+  std::set<std::string> GetNonAtomicTypeNames(
+      EntityTypeID entity_type_id) const {
+    std::set<std::string> result;
+    auto atomic_types = GetAtomicSubtypes(entity_type_id);
+    for (EntityTypeID super_type_id = 0; super_type_id < atomic_types.size();
+         ++super_type_id) {
+      if (atomic_types.test(super_type_id)) {
```

Comment:
I think I added iteration to `DynamicBitset` and friends. So you don't need to do this, you can use a for-each loop.